projects/plugins/protect/src/class-protect-status.php

```diff
@@ -0,0 +1,222 @@
+<?php
+/**
+ * Class to handle the Protect Status of Jetpack Protect
+ *
+ * @package automattic/jetpack-protect-plugin
+ */
+
+namespace Automattic\Jetpack\Protect;
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use Automattic\Jetpack\Plugins_Installer;
+use Automattic\Jetpack\Redirect;
+use Automattic\Jetpack\Sync\Functions as Sync_Functions;
+use Jetpack_Options;
+use WP_Error;
+
+/**
+ * Class that handles fetching and caching the Status of vulnerabilities check from the WPCOM servers
+ */
+class Protect_Status extends Status {
+
+	/**
+	 * WPCOM endpoint
+	 *
+	 * @var string
+	 */
+	const REST_API_BASE = '/sites/%d/jetpack-protect-status';
+
+	/**
+	 * Name of the option where status is stored
+	 *
+	 * @var string
+	 */
+	public static $option_name = 'jetpack_protect_status';
```

Comment:
Thanks so much for the insightful review @sergeymitr, I appreciate it!

I'm not extraordinarily familiar with abstract classes and methods in PHP, so that might ultimately be my answer to your question. But here is my reasoning for these changes:

I changed the use of constants to properties because every child of `Status` should likely be declaring a unique value for where it saves its particular type of status (`option_name`), how long to cache that status (`option_expires_after`), etc. Conceptually, this felt like a shift away from a value that is "constant" or defined once and towards a value that is rather designed to be defined a variable amount of times, by the child classes or potentially inherited in the case of the default cache timing values like `option_expires_after`.

Also, defining an empty constant i.e. `const OPTION_NAME = '';` in the primary `Status` class just felt wrong to me, which I believe is required to define those shared methods in the root `Status` class.

Would a commit such as https://github.com/Automattic/jetpack/pull/26119/commits/9bd82bd1c743a822d22a457a18236abc0f765205 resolve this specific comment, or are you imagining something more significant with your mention of abstract methods? I'm not exactly sure how those would relate to the definition of the status option names and timing.